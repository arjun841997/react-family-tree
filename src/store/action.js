import api from "../api";
import {NotificationManager} from 'react-notifications';
export const saveTree = (data) => async(dispatch) => {
    const response = await api.post('/family', data);
    NotificationManager.success('Family Tree Saved Successfully.', 'Success!!');
    dispatch({
        type: 'SAVE_TREE',
        payload: response
    });
}

export const getFamilies = (data) => async(dispatch) => {
    const response = await api.get('/family');
    dispatch({
        type: 'GET_FAMILIES',
        payload: response
    });
}