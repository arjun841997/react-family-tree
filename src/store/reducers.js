var initialState ={
    data: {
        data: []
    }
}

export const treeReducer = (state = initialState, action)  => {
   
    if (action.type === 'SAVE_TREE') {
        // return action.payload;
    }
    if(action.type === 'GET_FAMILIES'){
      return action.payload
    }
    return state;
}