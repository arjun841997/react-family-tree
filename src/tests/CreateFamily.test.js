import React from 'react';
import renderer from 'react-test-renderer';
import store from '../store'
import { Provider } from 'react-redux'

import CreateFamily from '../containers/createFamily'

it('Create Family Snapshot', () => {
  const tree = renderer.create(<Provider store={store}><CreateFamily /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});