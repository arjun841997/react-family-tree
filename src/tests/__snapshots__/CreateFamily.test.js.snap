// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Family Snapshot 1`] = `
Array [
  <div
    className="username mb-3 input-group"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
        id="basic-addon1"
      >
        Family Name
      </span>
    </div>
    <input
      className="form-control"
      id="familyName"
      onChange={[Function]}
      placeholder="Family Name"
    />
  </div>,
  <div
    className="username mb-3 input-group"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
        id="basic-addon1"
      >
        Family Head Name
      </span>
    </div>
    <input
      className="form-control"
      id="familyHeadName"
      onChange={[Function]}
      placeholder="Family Head Name"
    />
  </div>,
  <div
    className="username mb-3 input-group"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
      >
        Gender
      </span>
    </div>
    <input
      className="form-control"
      id="gender"
      onChange={[Function]}
      placeholder="Gender"
    />
  </div>,
  <button
    className="btn btn-light"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add Root Node
  </button>,
  <button
    className="btn btn-light"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Save
  </button>,
]
`;
